Yosemite/Yosemite/Stores/ProductStore.swift

```diff
@@ -512,6 +513,24 @@ private extension ProductStore {
         }
         storageProduct.addToTags(NSOrderedSet(array: storageTags))
     }
+
+    /// Updates, inserts, or prunes the provided StorageProduct's downloads using the provided read-only Product's downloads
+    ///
+    func handleProductDownloads(_ readOnlyProduct: Networking.Product, _ storageProduct: Storage.Product, _ storage: StorageType) {
+        // Removes all the downloads first.
+        storageProduct.downloads?.forEach { existingStorageImage in
+            storage.deleteObject(existingStorageImage)
+        }
+
+        // Inserts the downloads from the read-only product variation.
```

Comment:
product variation?

Reply:
I wrote it wrong. my bad. Also because at that time, I was not aware of the difference in product types.